### ISSUE TITLE
Relative links must end with the ".md" extension

### ISSUE THREAD

**Issue by mimig1** [craigshoemaker/spinefeed-function#9]
Jason Howell pointed out the following "I thought #anchors were allowed for links inside the same file without .md files mentioned. Also cross-repo to cli repo doesn’t allow .md to be mentioned since they are built separately." So I think we need some rule modifications here.

**Comment by JasonWHowell** [comment]
Example: 
https://github.com/MicrosoftDocs/azure-docs-pr/edit/master/articles/mariadb/scripts/sample-change-server-configuration.md

This rule is very broad, and doesn't list which link in the article, but the article has a ton of links. Assuming the CLI links are contentious because they are relative yet not markdown files.

> Relative links must end with the ".md" extension | C:/Users/cshoe/Documents/data/docs/az-docs-pr/articles/mariadb/scripts/sample-change-server-configuration.md | sample-change-server-configuration.md | sample | andrela | ajlam | 11/28/2018

These are links to reference docs that are not .md files. I think they are probably .yml, but to link outside the repo, you cannot use the file extension as the files are not part of the build process for the current repo. 

```
| [az group create](/cli/azure/group#az-group-create) | Creates a resource group in which all resources are stored. |
| [az mariadb server create](/cli/azure/mariadb/server#az-mariadb-server-create) | Creates a MariaDB server that hosts the databases. |
| [az mariadb server configuration list](/cli/azure/mariadb/server/configuration#az-mariadb-server-configuration-list) | List the configurations of an Azure Database for MariaDB server. |
| [az mariadb server configuration set](/cli/azure/mariadb/server/configuration#az-mariadb-server-configuration-set) | Update the configuration of an Azure Database for MariaDB server. |
| [az mariadb server configuration show](/cli/azure/mariadb/server/configuration#az-mariadb-server-configuration-show) | Show the configuration of an Azure Database for MariaDB server. |
| [az group delete](/cli/azure/group#az-group-delete) | Deletes a resource group including all nested resources. |
```

**Comment by craigshoemaker** [comment]
Closed by 7dd1a53439b6000bc1d0f861d4ac507795d6cbd9

This rule now strips off TOC location from a link before evaluation.
